pkg/clients/builder.go

```diff
@@ -47,14 +47,18 @@ func NewBuilder(userAgent string, csiDriverName string, controllerConfig *contro
 		guestNamespaces: []string{
 			"",
 			CSIDriverNamespace,
-			ManagedConfigNamespace, // TODO: is it needed?
 		},
 	}
 }
 
 // WithHyperShiftGuest sets the kubeconfig file for the guest cluster. Usable only on HyperShift.
-func (b *Builder) WithHyperShiftGuest(kubeConfigFile string) *Builder {
+func (b *Builder) WithHyperShiftGuest(kubeConfigFile string, cloudConfigNamespace string) *Builder {
 	b.guestKubeConfigFile = kubeConfigFile
+	if cloudConfigNamespace != "" {
+		b.guestNamespaces = append(b.guestNamespaces, cloudConfigNamespace)
+	} else {
+		b.guestNamespaces = append(b.guestNamespaces, ManagedConfigNamespace)
```

Comment:
I think we can remove the whole `ManagedConfigNamespace` variable and AWS EBS can use `CloudConfigNamespace: cloudConfigNamespace` (where cloudConfigNamespace is [this](https://github.com/openshift/csi-operator/blob/323d0d0032a21637ba98cd4bac0ef932bf147fe0/pkg/driver/aws-ebs/aws_ebs.go#L33))

Reply:
What about this comment? It's easy and it would make the common code a bit easier to read.